pkg/controllers/member/endpointslice/controller.go

```diff
@@ -110,6 +113,11 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, nil
 	}
 
+	if !r.joined.Load() {
+		klog.V(2).InfoS("EndpointSlice controller is not started yet, requeue the request", "endpointSlice", endpointSliceRef)
```

Comment:
Hi Zhiying! I think it might be better to add a comment here explaining that if the cluster has left the controller will continue performing the unexporting duties and this is intentional.

Reply:
The log might need to reflect this as well.

Reply:
Same for the other few applicable controllers I think.